### PR TITLE
task 9 done.

### PR DESCRIPTION
**Pull request by 1mb0**

### CODE REVIEW

File: src/main/java/md/jack/task9/Primes.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+ * and find the next prime number above a given BigInteger.
+ * <p>
+ * From <a href="http://courses.coreservlets.com/Course-Materials/">the
+ * coreservlets.com tutorials on JSF 2, PrimeFaces, Ajax, JavaScript, jQuery, GWT, Android,
+ * Spring, Hibernate, JPA, RESTful Web Services, Hadoop, Spring MVC,
+ * servlets, JSP, Java 8 lambdas and streams (for those that know Java already),
+ * and Java 8 programming (for those new to Java)</a>.
+ */
+
+public class Primes
```
Comment (jackcucu) [inline src/main/java/md/jack/task9/Primes.java]: Util classes has to be declared as final

File: src/main/java/md/jack/task9/Primes.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+     * Generates a random number of the given length,
+     * then finds the first prime number above that random number.
+     */
+    public static BigInteger findPrime(int numDigits)
+    {
+        if (numDigits < 1)
+        {
+            numDigits = 1;
+        }
+        return (nextPrime(randomNum(numDigits)));
```
Comment (jackcucu) [inline src/main/java/md/jack/task9/Primes.java]: do not use unnecessary braces in return form

File: src/main/java/md/jack/task9/Primes.java [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+     * cannot be a zero).
+     */
+
+    public static BigInteger randomNum(int numDigits)
+    {
+        StringBuilder s = new StringBuilder();
+        // First digit must be non-zero.
+        s.append(randomDigit(false));
+
+        Stream.of(numDigits)
```
Comment (jackcucu) [inline src/main/java/md/jack/task9/Primes.java]: How does this stream work ?

File: src/main/java/md/jack/task9/Primes.java [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+    /**
+     * Creates a random big integer where every digit is
+     * selected randomly (except that the first digit
+     * cannot be a zero).
+     */
+
+    public static BigInteger randomNum(int numDigits)
+    {
+        StringBuilder s = new StringBuilder();
```
Comment (jackcucu) [inline src/main/java/md/jack/task9/Primes.java]: Local vars has to be declared as final

File: src/main/java/md/jack/task9/Primes.java [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    public static boolean isPrime(long possiblePrime)
+    {
+        return (isPrime(new BigInteger(Long.toString(possiblePrime))));
+    }
+
+    /** Finds the next prime number above a threshold. */
+
+    private static BigInteger nextPrime(BigInteger start)
```
Comment (jackcucu) [inline src/main/java/md/jack/task9/Primes.java]: method args has to be declared as final, try to rewrite this method after declare *start* object as final

File: src/main/java/md/jack/task9/Primes.java [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+            return (start);
+        }
+        return (nextPrime(start));
+    }
+
+    /**
+     * Generates a random number of the given length,
+     * then finds the first prime number above that random number.
+     */
+    public static BigInteger findPrime(int numDigits)
```
Comment (jackcucu) [inline src/main/java/md/jack/task9/Primes.java]: final method args

File: src/main/java/md/jack/task9/Primes.java [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    }
+
+    private static boolean isEven(BigInteger n)
+    {
+        return (n.mod(TWO).equals(ZERO));
+    }
+
+    private static String randomDigit(boolean isZeroOk)
+    {
+        if (isZeroOk)
```
Comment (jackcucu) [inline src/main/java/md/jack/task9/Primes.java]: try to replace if with ternary if https://alvinalexander.com/java/edu/pj/pj010018

File: src/main/java/md/jack/task9/Utils.java [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+    }
+
+    public static int countPrime(int upperBound)
+    {
+        return (primes.computeIfAbsent(upperBound, Utils::apply));
+    }
+
+    private static List<Double> randomList(int size)
+    {
+        final List<Double> nums = new ArrayList<>(size);
```
Comment (jackcucu) [inline src/main/java/md/jack/task9/Utils.java]: Why you didn't collect stream into a list ?

File: src/main/java/md/jack/task9/Primes.java (hunk 3 above)
Comment (1mb0) [reply src/main/java/md/jack/task9/Primes.java]: for (int i = 0; i < numDigits - 1; i++)
        {
            s.append(randomDigit(true));
        }
        return (new BigInteger(s.toString()));
something like this , or that's wrong ?

**Review by jackcucu — APPROVED** [review]

**Pull request merged**